### DISCUSSION TITLE
Restrict available parent entries when editing entry in Craft 4

### DISCUSSION THREAD

**Discussion by paulgoodfield** [craftcms/cms#12475] [Ideas] [answered]
Trying to work out how to achieve something in Craft 4 that I had working in version 3:

In a module I’m building I need to restrict which entries out of a structure are available to select as a parent when creating/editing an entry. In Craft 3 I could hook into the edit screen and modify the parentOptionCriteria like this like:

```php
Craft::$app->view->hook('cp.entries.edit', function( array &$context )
{
    $context['parentOptionCriteria'] = [
        'siteId' => 1,
        'sectionId' => $client->section->id,
        'status' => null,
        'where' => [
            'in',
            'elements.id',
            $availableParentIds
        ],
        'drafts' => null,
        'draftOf' => false,
        'level' => '<= 2'
    ];
}
```

I understand that in Craft 4 this hook isn’t available as you’re now using the Unified Element Editor, but I can’t find any examples of how I could achieve the same functionality in this version. Any ideas?

**Comment by brandonkelly** [comment, marked as answer]
Just added a new `EVENT_DEFINE_PARENT_SELECTION_CRITERIA` event to `craft\elements\Entry` for Craft 4.4 (#12485). With it you’ll be able to do this:

```php
use craft\elements\Entry;
use craft\events\ElementCriteriaEvent;
use yii\base\Event;

Event::on(
    Entry::class,
    Entry::EVENT_DEFINE_PARENT_SELECTION_CRITERIA,
    function(ElementCriteriaEvent $event) {
        $event->criteria['id'] = $availableParentId;
        $event->criteria['level'] = '<= 2';
    },
);
```

**Reply by paulgoodfield** [reply]
Many thanks @brandonkelly ! Will give it a try asap.

**Reply by brandonkelly** [reply]
Craft 4.4 is out!
